src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -381,7 +381,7 @@ pub async fn merge_imms_in_memory(
         UserKey::new(table_id, TableKey(pivot.as_ref())),
         HummockEpoch::MAX,
     );
-    let mut versions: Vec<(HummockEpoch, HummockValue<Bytes>)> = Vec::new();
+    let mut versions: Vec<(EpochWithGap, HummockValue<Bytes>)> = Vec::new();
```

Comment:
We should change the return type of `current_items` to epoch with gap, and then the epoch in items should also be epoch with gap.

And in the following versions.push we should use the original epoch with gap in the item instead of created from new from epoch.